iris_grib/message.py

```diff
@@ -404,12 +412,22 @@ def __getitem__(self, key):
         if key not in self._cache:
             if key == 'numberOfSection':
                 value = self._number
-            elif key not in self._keys:
-                raise KeyError('{!r} not defined in section {}'.format(
-                    key, self._number))
             else:
+                keys = [key]
+                # Also check known alias alternatives
+                if key in KEY_ALIAS:
+                    keys.append(KEY_ALIAS[key])
+                keys = np.array(keys)
+                found = [k in self._keys for k in keys]
+                if not any(found):
+                    emsg = f"{key} not defined in section {self._number}"
+                    raise KeyError(emsg)
+                # Take the first valid key, original or alias.
+                key = keys[found][0]
```

Comment:
See this [comment](https://github.com/SciTools/iris-grib/pull/361/files#r1394257805)